zerver/views/streams.py

```diff
@@ -454,7 +454,7 @@ def get_subscription_or_die(stream_name, user_profile):
     # type: (text_type, UserProfile) -> Subscription
     stream = get_stream(stream_name, user_profile.realm)
     if not stream:
-        raise JsonableError(_("Invalid stream %s") % (stream.name,))
```

Comment:
Since this is the one place of your series of commits where you had to introduce a small change to the "real" code, I would try to isolate which test here is the failing test that exposes that issue, and then put the one-line change and the failing test in a small commit.  (Of course, it wouldn't be a failing test at that point--I hope you not what I'm clumsily trying to say here.)

Then split out tests that weren't affected here into another commit.

For the tests unaffected by this small code change, your commit message should be something along the lines of `Increase test coverage of streams API.`

For the smaller commit that introduces the fix, your commit will be something like `Fix "Invalid stream" error message in get_subscription_or_die().`